### PR TITLE
feat: add intMax

### PR DESCRIPTION
**Pull request by bollu**

### CODE REVIEW

**Comment by bollu** [comment]
awaiting-review

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -599,4 +599,20 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   simp
   exact Nat.lt_of_le_of_ne
 
+/- ! ### intMax -/
+
+/-- A bitvector when interpreted as an integer is the least value: -2^(w-1) when w > 0-/
+def intMax (w : Nat) : BitVec w  := (2^w - 1)#w
+
+theorem intMax_getLsb_eq (w : Nat) : (intMax w).getLsb i = decide (i < w) := by
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:607]: Should be `getLsb_intMax`, also below.

File: src/Init/Data/BitVec/Lemmas.lean [hunk 2]
```diff
@@ -599,4 +599,20 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   simp
   exact Nat.lt_of_le_of_ne
 
+/- ! ### intMax -/
+
+/-- A bitvector when interpreted as an integer is the least value: -2^(w-1) when w > 0-/
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:604]:
```suggestion
/-- A bitvector when interpreted as an integer is the least value: -2^(w-1) when w > 0-/
```
Garbled?

File: src/Init/Data/BitVec/Lemmas.lean [hunk 3]
```diff
@@ -599,4 +599,20 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
...
+def intMax (w : Nat) : BitVec w  := (2^w - 1)#w
+
+theorem intMax_getLsb_eq (w : Nat) : (intMax w).getLsb i = decide (i < w) := by
+  simp [intMax, getLsb]
+
+theorem intMax_toNat_eq : (intMax w).toNat = if w = 0 then 0 else 2^w - 1 := by
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:610]: No `if` required, 2^w - 1 is correct always.

File: src/Init/Data/BitVec/Lemmas.lean (hunk 2 above)
Comment (bollu) [reply src/Init/Data/BitVec/Lemmas.lean]: Thanks, fixed.

**Comment by bollu** [comment]
awaiting-review

File: src/Init/Data/BitVec/Lemmas.lean (hunk 1 above)
Comment (bollu) [reply src/Init/Data/BitVec/Lemmas.lean]: done.

**Review by joehendrix — APPROVED** [review]

**Pull request merged**